### PR TITLE
Determine sequencing parameters automatically for Illumina runs

### PR DESCRIPTION
**Pull request by apmasell**

### CODE REVIEW

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/IlluminaNotificationDto.java [hunk 1]
```diff
@@ -66,17 +88,28 @@ public void setNumCycles(int numCycles) {
...
+  public void setReadLength(int readLength) {
+    this.readLength = readLength;
+  }
+
   public void setScoreCycle(int scoreCycle) {
     this.scoreCycle = scoreCycle;
   }
 
   @Override
+  public boolean test(SequencingParameters params) {
```
Comment (callunity) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/IlluminaNotificationDto.java]: Perhaps a more descriptive name?

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java [hunk 2]
```diff
@@ -182,4 +184,9 @@ public boolean equals(Object obj) {
   public Optional<String> getLaneContents(int lane) {
```
Comment (sqwci) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java]: MAJOR Remove this unused method parameter "lane". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/SequencingParameters.java [hunk 3]
```diff
@@ -3,45 +3,150 @@
...
+  private Platform platform;
+
+  private int readLength;
+
+  @ManyToOne(targetEntity = UserImpl.class, fetch = FetchType.LAZY)
+  @JoinColumn(name = "updatedBy", nullable = false)
+  private User updatedBy;
+
+  @Override
+  public int compareTo(SequencingParameters o) {
```
Comment (sqwci) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/SequencingParameters.java]: MINOR Override "equals(Object obj)" to comply with the contract of the "compareTo(T o)" method. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1210)

File: runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java [hunk 4]
```diff
@@ -52,6 +58,16 @@ public NotificationDto process(File runDirectory, TimeZone tz) throws IOExceptio
...
+      Document parameters = DocumentBuilderFactory.newInstance().newDocumentBuilder().parse(new File(runDirectory, "runParameters.xml"));
+      dto.setChemistry(Arrays.stream(IlluminaChemistry.values()).filter(chemistry -> chemistry.test(parameters)).findFirst()
+          .orElse(IlluminaChemistry.UNKNOWN));
+    } catch (SAXException | ParserConfigurationException e) {
+      log.error("Failed to parse parameters", e);
+    }
+
     // The Illumina library can't distinguish between a failed run and one that either finished or is still going. Scan the logs, if
     // available to determine if the run failed.
     File rtaLogDir = new File(runDirectory, "/Data/RTALogs");
```
Comment (sqwci) [inline runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java]: MINOR Refactor your code to get this URI from a customizable parameter. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1075)

**Comment by sqwci** [comment]
SonarQube analysis reported 11 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [Dtos.java#L605](https://github.com/TGAC/miso-lims/blob/d8a8ff7fe2cd2a36e940c14f1295c2f425170cdf/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L605): Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [Dtos.java#L1727](https://github.com/TGAC/miso-lims/blob/d8a8ff7fe2cd2a36e940c14f1295c2f425170cdf/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L1727): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)
1. ![MAJOR][MAJOR] [Dtos.java#L1727](https://github.com/TGAC/miso-lims/blob/d8a8ff7fe2cd2a36e940c14f1295c2f425170cdf/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L1727): Remove this unused method parameter "from". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L37](https://github.com/TGAC/miso-lims/blob/d8a8ff7fe2cd2a36e940c14f1295c2f425170cdf/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L37): Remove this unused method parameter "parameters". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L47](https://github.com/TGAC/miso-lims/blob/d8a8ff7fe2cd2a36e940c14f1295c2f425170cdf/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L47): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L57](https://github.com/TGAC/miso-lims/blob/d8a8ff7fe2cd2a36e940c14f1295c2f425170cdf/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L57): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)
1. ![MINOR][MINOR] [RunScannerClient.java#L10](https://github.com/TGAC/miso-lims/blob/d8a8ff7fe2cd2a36e940c14f1295c2f425170cdf/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/service/RunScannerClient.java#L10): Remove this unused import 'java.util.function.Predicate'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [RunScannerClient.java#L25](https://github.com/TGAC/miso-lims/blob/d8a8ff7fe2cd2a36e940c14f1295c2f425170cdf/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/service/RunScannerClient.java#L25): Remove this unused import 'uk.ac.bbsrc.tgac.miso.core.data.SequencingParameters'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/IlluminaNotificationDto.java (hunk 1 above)
Comment (apmasell) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/IlluminaNotificationDto.java]: It's implied by the predicate interface.

File: sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateSequencingParametersDao.java [hunk 5]
```diff
@@ -46,7 +45,7 @@ public SequencingParameters getSequencingParameters(Long id) throws IOException
     if (id == null) {
       return null;
     }
-    return (SequencingParameters) currentSession().get(SequencingParametersImpl.class, id);
+    return (SequencingParameters) currentSession().get(SequencingParameters.class, id);
   }
```
Comment (callunity) [inline sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateSequencingParametersDao.java]: Please make changes to `getSequencingParameters()` as well; it's still looking for the Impl.

Comment (apmasell) [inline sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateSequencingParametersDao.java]: I made it a criteria.

**Comment by sqwci** [comment]
SonarQube analysis reported 11 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [Dtos.java#L605](https://github.com/TGAC/miso-lims/blob/d1c8b791bd2b59588dc7fd8d740a2d74e778afc7/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L605): Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [Dtos.java#L1727](https://github.com/TGAC/miso-lims/blob/d1c8b791bd2b59588dc7fd8d740a2d74e778afc7/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L1727): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)
1. ![MAJOR][MAJOR] [Dtos.java#L1727](https://github.com/TGAC/miso-lims/blob/d1c8b791bd2b59588dc7fd8d740a2d74e778afc7/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L1727): Remove this unused method parameter "from". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L37](https://github.com/TGAC/miso-lims/blob/d1c8b791bd2b59588dc7fd8d740a2d74e778afc7/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L37): Remove this unused method parameter "parameters". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L47](https://github.com/TGAC/miso-lims/blob/d1c8b791bd2b59588dc7fd8d740a2d74e778afc7/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L47): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L57](https://github.com/TGAC/miso-lims/blob/d1c8b791bd2b59588dc7fd8d740a2d74e778afc7/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L57): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)
1. ![MINOR][MINOR] [RunScannerClient.java#L10](https://github.com/TGAC/miso-lims/blob/d1c8b791bd2b59588dc7fd8d740a2d74e778afc7/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/service/RunScannerClient.java#L10): Remove this unused import 'java.util.function.Predicate'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [RunScannerClient.java#L25](https://github.com/TGAC/miso-lims/blob/d1c8b791bd2b59588dc7fd8d740a2d74e778afc7/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/service/RunScannerClient.java#L25): Remove this unused import 'uk.ac.bbsrc.tgac.miso.core.data.SequencingParameters'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/IlluminaNotificationDto.java [hunk 6]
```diff
@@ -48,16 +64,22 @@ public int hashCode() {
     final int prime = 31;
```
Comment (sqwci) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/IlluminaNotificationDto.java]: CRITICAL Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)

**Comment by sqwci** [comment]
SonarQube analysis reported 14 issues
* ![CRITICAL][CRITICAL] 5 critical
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [PoolOrderCompletion.java#L71](https://github.com/TGAC/miso-lims/blob/340116cfd4e3a28544eca9d3568756a877aeb982/core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/PoolOrderCompletion.java#L71): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![CRITICAL][CRITICAL] [Dtos.java#L605](https://github.com/TGAC/miso-lims/blob/340116cfd4e3a28544eca9d3568756a877aeb982/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L605): Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [Dtos.java#L1727](https://github.com/TGAC/miso-lims/blob/340116cfd4e3a28544eca9d3568756a877aeb982/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L1727): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)
1. ![CRITICAL][CRITICAL] [NotificationDto.java#L132](https://github.com/TGAC/miso-lims/blob/340116cfd4e3a28544eca9d3568756a877aeb982/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java#L132): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![MAJOR][MAJOR] [Dtos.java#L1727](https://github.com/TGAC/miso-lims/blob/340116cfd4e3a28544eca9d3568756a877aeb982/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L1727): Remove this unused method parameter "from". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L37](https://github.com/TGAC/miso-lims/blob/340116cfd4e3a28544eca9d3568756a877aeb982/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L37): Remove this unused method parameter "parameters". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L47](https://github.com/TGAC/miso-lims/blob/340116cfd4e3a28544eca9d3568756a877aeb982/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L47): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L57](https://github.com/TGAC/miso-lims/blob/340116cfd4e3a28544eca9d3568756a877aeb982/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L57): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)
1. ![MINOR][MINOR] [RunScannerClient.java#L10](https://github.com/TGAC/miso-lims/blob/340116cfd4e3a28544eca9d3568756a877aeb982/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/service/RunScannerClient.java#L10): Remove this unused import 'java.util.function.Predicate'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [RunScannerClient.java#L25](https://github.com/TGAC/miso-lims/blob/340116cfd4e3a28544eca9d3568756a877aeb982/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/service/RunScannerClient.java#L25): Remove this unused import 'uk.ac.bbsrc.tgac.miso.core.data.SequencingParameters'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 14 issues
* ![CRITICAL][CRITICAL] 5 critical
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [PoolOrderCompletion.java#L71](https://github.com/TGAC/miso-lims/blob/ad19951aff89894275cdc440f07224af54132e61/core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/PoolOrderCompletion.java#L71): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![CRITICAL][CRITICAL] [Dtos.java#L605](https://github.com/TGAC/miso-lims/blob/ad19951aff89894275cdc440f07224af54132e61/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L605): Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [Dtos.java#L1727](https://github.com/TGAC/miso-lims/blob/ad19951aff89894275cdc440f07224af54132e61/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L1727): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)
1. ![CRITICAL][CRITICAL] [NotificationDto.java#L132](https://github.com/TGAC/miso-lims/blob/ad19951aff89894275cdc440f07224af54132e61/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java#L132): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![MAJOR][MAJOR] [Dtos.java#L1727](https://github.com/TGAC/miso-lims/blob/ad19951aff89894275cdc440f07224af54132e61/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L1727): Remove this unused method parameter "from". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L37](https://github.com/TGAC/miso-lims/blob/ad19951aff89894275cdc440f07224af54132e61/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L37): Remove this unused method parameter "parameters". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L47](https://github.com/TGAC/miso-lims/blob/ad19951aff89894275cdc440f07224af54132e61/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L47): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L57](https://github.com/TGAC/miso-lims/blob/ad19951aff89894275cdc440f07224af54132e61/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L57): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)
1. ![MINOR][MINOR] [RunScannerClient.java#L10](https://github.com/TGAC/miso-lims/blob/ad19951aff89894275cdc440f07224af54132e61/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/service/RunScannerClient.java#L10): Remove this unused import 'java.util.function.Predicate'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [RunScannerClient.java#L25](https://github.com/TGAC/miso-lims/blob/ad19951aff89894275cdc440f07224af54132e61/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/service/RunScannerClient.java#L25): Remove this unused import 'uk.ac.bbsrc.tgac.miso.core.data.SequencingParameters'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 12 issues
* ![CRITICAL][CRITICAL] 5 critical
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [PoolOrderCompletion.java#L71](https://github.com/TGAC/miso-lims/blob/328c6acb4c5a49780400dfa3c3032e4d2cf4adc9/core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/PoolOrderCompletion.java#L71): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![CRITICAL][CRITICAL] [Dtos.java#L605](https://github.com/TGAC/miso-lims/blob/328c6acb4c5a49780400dfa3c3032e4d2cf4adc9/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L605): Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [Dtos.java#L1727](https://github.com/TGAC/miso-lims/blob/328c6acb4c5a49780400dfa3c3032e4d2cf4adc9/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L1727): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)
1. ![CRITICAL][CRITICAL] [NotificationDto.java#L132](https://github.com/TGAC/miso-lims/blob/328c6acb4c5a49780400dfa3c3032e4d2cf4adc9/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java#L132): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![MAJOR][MAJOR] [Dtos.java#L1727](https://github.com/TGAC/miso-lims/blob/328c6acb4c5a49780400dfa3c3032e4d2cf4adc9/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L1727): Remove this unused method parameter "from". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L37](https://github.com/TGAC/miso-lims/blob/328c6acb4c5a49780400dfa3c3032e4d2cf4adc9/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L37): Remove this unused method parameter "parameters". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L47](https://github.com/TGAC/miso-lims/blob/328c6acb4c5a49780400dfa3c3032e4d2cf4adc9/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L47): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L57](https://github.com/TGAC/miso-lims/blob/328c6acb4c5a49780400dfa3c3032e4d2cf4adc9/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L57): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 9 issues
* ![CRITICAL][CRITICAL] 4 critical
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [PoolOrderCompletion.java#L71](https://github.com/TGAC/miso-lims/blob/9979cb7515e3332282fac109921c531014dd5d92/core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/PoolOrderCompletion.java#L71): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![CRITICAL][CRITICAL] [Dtos.java#L604](https://github.com/TGAC/miso-lims/blob/9979cb7515e3332282fac109921c531014dd5d92/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L604): Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [NotificationDto.java#L132](https://github.com/TGAC/miso-lims/blob/9979cb7515e3332282fac109921c531014dd5d92/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java#L132): Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00115)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L37](https://github.com/TGAC/miso-lims/blob/9979cb7515e3332282fac109921c531014dd5d92/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L37): Remove this unused method parameter "parameters". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L47](https://github.com/TGAC/miso-lims/blob/9979cb7515e3332282fac109921c531014dd5d92/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L47): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [StandardIllumina.java#L57](https://github.com/TGAC/miso-lims/blob/9979cb7515e3332282fac109921c531014dd5d92/runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java#L57): Either re-interrupt this method or rethrow the "InterruptedException". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**